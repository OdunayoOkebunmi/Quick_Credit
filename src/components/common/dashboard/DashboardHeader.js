import React from 'react';
import { Link } from 'react-router-dom';

const DashboardHeader = () => (
  <header className="dashboard-header">
    <Link to="/">
      <img
        srcSet="https://res.cloudinary.com/dnl5kwhe6/image/upload/v1567563417/Quick_Credit/logo-sm_o2xdjf.png 1x, https://res.cloudinary.com/dnl5kwhe6/image/upload/v1567563417/Quick_Credit/logo-md_a1owld.png 2x"
        alt="Quick Credit logo"
        src="https://res.cloudinary.com/dnl5kwhe6/image/upload/v1567563417/Quick_Credit/logo-sm_o2xdjf.png"
        className="logo"
      />
    </Link>
    <form action="" className="search">
      <input type="text" className="search__input" placeholder="Search" />
      <button className="search__button" type="button">
        <i className="search__icon fas fa-search" />
      </button>
    </form>
    <nav className="user-nav">
      <div className="user-nav__user">
        <span className="user-nav__user-name">Jane Doe</span>
      </div>
    </nav>
  </header>
);

export default DashboardHeader;
