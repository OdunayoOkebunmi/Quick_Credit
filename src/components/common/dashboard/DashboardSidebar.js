/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';
import { userLogOut } from '@Actions/authActions';

class DashboardSidebar extends Component {
  logOut = () => {
    const { quit, history } = this.props;
    return quit(history);
  }

  render() {
    return (
      <nav className="sidebar">
        <ul className="side-nav">
          <li className="side-nav__item side-nav__item--active">
            <a href="user-dashboard.html" className="side-nav__link">
              <i className="side-nav__icon fas fa-tachometer-alt" />
              <span>Dashboard</span>
            </a>
          </li>
          <li className="side-nav__item">
            <Link to="/" className="side-nav__link">
              <i className="side-nav__icon fas fa-edit" />

              <span>Apply for Loans</span>
            </Link>
          </li>
          <li className="side-nav__item">
            <Link to="/profile" className="side-nav__link">
              <i className="side-nav__icon fas fa-user" />
              <span>Profile</span>
            </Link>
          </li>
          <li className="side-nav__item" id="logout">
            <Link to="/" className="side-nav__link" onClick={this.logOut}>
              <i className="side-nav__icon fas fa-sign-out-alt" />
              <span>Logout</span>
            </Link>
          </li>
        </ul>
      </nav>
    );
  }
}
DashboardSidebar.propTypes = {
  history: PropTypes.object.isRequired,
  quit: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  quit: history => dispatch(userLogOut(history)),
});
export default connect(null, mapDispatchToProps)(withRouter(DashboardSidebar));
