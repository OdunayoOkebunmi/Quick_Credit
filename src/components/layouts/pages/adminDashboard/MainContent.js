import React from 'react';

const MainContent = () => (
  <main className="admin-users-loans--view">
    <div className="admin-users-loans__details">
      <h3 className="heading-secondary">
        loan applications
      </h3>
      <table className="admin-users-loans__details--table">
        <thead>
          <tr>
            <th>Loan ID</th>
            <th>Email</th>
            <th>Date Created</th>
            <th>Loan Status</th>
            <th>Repaid</th>
            <th>Tenor</th>
            <th>Amount ₦</th>
            <th>
              Monthly
              {' '}
              <br />
              Installments
              {' '}
              <br />
              ₦
            </th>
            <th>
              Balance
              {' '}
              <br />
              ₦
            </th>
            <th>
              Interest
              {' '}
              <br />
              ₦
            </th>
            <th>Actions</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>#122392</td>
            <td>odun@mail.com</td>
            <td>2019/01/01</td>
            <td>Pending</td>
            <td>False</td>
            <td>10</td>
            <td>2,000,000</td>
            <td>210,000</td>
            <td>2,000,000</td>
            <td>100,000</td>
            <td>
              <a
                href="#user-loan-popup"
                title="View Single Loan"
              >
                <i className="fa fa-eye" />

              </a>
              <a href="#">
                <i className="fa fa-check" title="Approve Loan" />

              </a>
              <a href="#">
                <i className="fa fa-times" title="Reject Loan" />

              </a>
            </td>
          </tr>

          <tr>
            <td>#122323</td>
            <td>harry@hedwig.com</td>
            <td>2019/01/01</td>
            <td>Approved</td>
            <td>True</td>
            <td>12</td>
            <td>150,000</td>
            <td>13,125</td>
            <td>0</td>
            <td>7,500</td>
            <td>
              <a
                href="#user-loan-popup"
                title="View Single Loan"
              >
                <i className="fa fa-eye" />

              </a>
              <a href="#">
                <i className="fa fa-check" title="Approve Loan" />

              </a>
              <a href="#">
                <i className="fa fa-times" title="Reject Loan" />

              </a>
            </td>
          </tr>
          <tr>
            <td>#122323</td>
            <td>gandalf@gray.com</td>
            <td>2018/02/11</td>
            <td>Declined</td>
            <td>False</td>
            <td>4</td>
            <td>100,000</td>
            <td>30,000</td>
            <td>100,000</td>
            <td>5000</td>
            <td>
              <a
                href="#user-loan-popup"
                title="View Single Loan"
              >
                <i className="fa fa-eye" />

              </a>
              <a href="#">
                <i className="fa fa-check" title="Approve Loan" />

              </a>
              <a href="#">
                <i className="fa fa-times" title="Reject Loan" />

              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </main>
);

export default MainContent;
