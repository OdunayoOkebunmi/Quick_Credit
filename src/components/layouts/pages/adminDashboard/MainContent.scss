
/* ++++++++STYLING FOR ADMIN USERS-LOAN APPLICATION VIEW SECTION +++++++ */
.content .admin-users-loans--view {
  flex: 1;
  font-size: 1.3rem;
  padding: 7rem 0;
  overflow-x: auto;
}
@media only screen and (max-width: 600px) {
  .content .admin-users-loans--view {
    padding: 3rem 0;
  }
}

@media only screen and (max-width: 900px) {
  .admin-users-loans__details .caption {
    font-size: 3rem;
  }
}
@media only screen and (max-width: 600px) {
  .admin-users-loans__details .caption {
    font-size: 2.5rem;
  }
}
.admin-users-loans--view .admin-users-loans__details--table {
  border-collapse: collapse;
  width: 100%;
  margin-left: 1rem;
}

.admin-users-loans__details--table td,
.admin-users-loans__details--table th {
  border: 1px solid #ddd;
  padding: 0.8rem;
  height: 2rem;
}

.admin-users-loans__details--table td {
  height: 5rem;
}
.admin-users-loans__details--table td a {
  display: inline-block;
  text-decoration: none;
  padding: 0.5rem;
}
.admin-users-loans__details--table td a .fa-eye {
  color: var(--color-grey-dark-2);
}
.admin-users-loans__details--table td a .fa-check {
  color: green;
}
.admin-users-loans__details--table td a .fa-times {
  color: red;
}
@media only screen and (max-width: 600px) {
  .admin-users-loans__details--table td a {
    background-color: var(--color-primary);
    padding: 1rem 2rem;
  }
}
.admin-users-loans__details--table tr:nth-child(even) {
  background-color: #fff;
}

.admin-users-loans__details--table tr:hover {
  background-color: var(--color-grey-light-2);
}

.admin-users-loans__details--table th {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: left;
  background-color: var(--color-primary);
  color: #fff;
}
