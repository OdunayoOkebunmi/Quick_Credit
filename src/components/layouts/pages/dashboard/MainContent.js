import React from 'react';

const MainContent = () => (
  <main className="user-dashboard-view">
    <div className="user-dashboard__history">
      <h3 className="heading-secondary">
        Transaction details
      </h3>
      <table className="user-dashboard__history--table">
        <thead>
          <tr>
            <th>Loan ID</th>
            <th>Date Created</th>
            <th>Monthly Installments ₦</th>
            <th>Total Amount ₦</th>
            <th>Action</th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </main>
);

export default MainContent;
