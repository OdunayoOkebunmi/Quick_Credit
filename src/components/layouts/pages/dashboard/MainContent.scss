
/* ++++++++STYLING FOR USER DASHBOARD VIEW SECTION +++++++ */
.content .user-dashboard-view {
  flex: 1;
  font-size: 1.2rem;
  padding: 7rem 0;
  overflow-x: auto;
  background-color: var(--color-grey-light-2);

}
@media only screen and (max-width: 600px) {
  .content .user-dashboard-view {
    padding: 3rem 0;
  }
}

@media only screen and (max-width: 900px) {
  .user-dashboard__history .caption {
    font-size: 3rem;
  }
}
@media only screen and (max-width: 600px) {
  .user-dashboard__history .caption {
    font-size: 2.5rem;
  }
}
.user-dashboard-view .user-dashboard__history--table {
  border-collapse: collapse;
  width: 100%;
}

.user-dashboard__history--table td,
.user-dashboard__history--table th {
  border: 1px solid #ddd;
  padding: 0.8rem;
  height: 2rem;
}

.user-dashboard__history--table td {
  height: 5rem;
}
.user-dashboard__history--table td a {
  text-decoration: none;
  padding: 1rem 2.5rem;
  color: #fff;
  background-color: var(--color-grey-dark-2);
}
.user-dashboard__history--table tr:nth-child(even) {
  background-color: #fff;
}

.user-dashboard__history--table tr:hover {
  background-color: var(--color-grey-light-3);
}

.user-dashboard__history--table th {
  padding-top: 12px;
  padding-bottom: 12px;
  text-align: left;
  background-color: var(--color-primary);
  color: white;
}

/* MEDIA QUERY TO MAKE TABLE RESPONSIVE */
@media screen and (max-width: 600px) {
  .user-dashboard__history--table > * > * > * {
    display: block;
  }
  .user-dashboard__history--table thead {
    display: none;
  }

  td {
    border: none;
    position: relative;
    padding-left: 20rem;
    margin-left: 22rem;
    font-size: 1.2rem;
  }
  td::before {
    position: absolute;
    top: 1.2rem;
    left: 0.6rem;
    width: 20rem;
    padding-right: 4rem;
    white-space: nowrap;
    margin-left: -22rem;
  }
  td::before {
    font-size: 1.2rem;
    text-transform: uppercase;
    font-weight: bolder;
  }
  td:nth-of-type(1):before {
    content: "Loan ID";
  }
  td:nth-of-type(2):before {
    content: "Date Created";
  }
  td:nth-of-type(3):before {
    content: "Monthly Installments";
  }
  td:nth-of-type(4):before {
    content: "Amount";
  }
}
