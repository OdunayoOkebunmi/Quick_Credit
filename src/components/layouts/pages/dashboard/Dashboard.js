/* eslint-disable no-console */
import React from 'react';
import DashboardHeader from '@Common/dashboard/DashboardHeader';
import DashboardSidebar from '@Common/dashboard/DashboardSidebar';
import MainContent from './MainContent';
import '@Common/styles/dashboard.scss';
import './MainContent.scss';


const Dashboard = () => (
  <div className="dashboard-container">
    <DashboardHeader />
    <div className="content">
      <DashboardSidebar />
      <MainContent />
    </div>
  </div>
);


export default Dashboard;
