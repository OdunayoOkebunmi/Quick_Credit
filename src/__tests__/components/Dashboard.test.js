import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import Dashboard from '@Layouts/pages/dashboard/Dashboard';
import DashboardHeader from '@Common/dashboard/DashboardHeader';
import DashboardSidebar from '@Common/dashboard/DashboardSidebar';
import MainContent from '@Layouts/pages/dashboard/MainContent';


const initialState = {
  auth: {
    error: null,
    user: {},
    isAuthenticated: false,
    status: 'status',
  },
};
const props = {
  history: {
    push: jest.fn(),
  },
};
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore(initialState);

describe('<DashboardSidebar/>', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <Router>
        <Dashboard store={store} {...props} />
      </Router>,
    );
    expect(component).toMatchSnapshot();
  });
});
it('should render without crashing', () => {
  mount(
    <Router>
      <MainContent />
    </Router>,
  );
});
describe('<DashboardSidebar/>', () => {
  it('should render without crashing', () => {
    mount(
      <Router>
        <DashboardHeader />
      </Router>,
    );
  });
});
describe('<DashboardSidebar/>', () => {
  it('should render without crashing', () => {
    mount(
      <Router>
        <DashboardSidebar store={store} {...props} />
      </Router>,
    );
  });
  it(' should call onSubmit prop function when form is submitted', () => {
    const logOut = jest.fn();
    const component = mount(
      <Router>
        <DashboardSidebar store={store} onClick={logOut} {...props} />
      </Router>,
    );
    const form = component.find('Link').at(2);
    form.simulate('click');
    expect(logOut).toHaveBeenCalledTimes(0);
    component.unmount();
  });
});
