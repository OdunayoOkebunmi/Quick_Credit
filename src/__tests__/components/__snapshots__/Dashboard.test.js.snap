// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardSidebar/> should render without crashing 1`] = `ShallowWrapper {}`;
